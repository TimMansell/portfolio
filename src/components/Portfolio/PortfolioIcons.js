import React from 'react';

import './PortfolioIcons.scss';

export const PortfolioIcons = () => {
  return <div className="portfolio-icons">
    <div className="portfolio-icons__icon" />
    <div className="portfolio-icons__icon" />
    <div className="portfolio-icons__icon" />
  </div>;
};

export default PortfolioIcons;
