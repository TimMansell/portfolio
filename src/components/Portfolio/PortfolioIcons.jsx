import React from 'react';
import PropTypes from 'prop-types';

import styles from './PortfolioIcons.module.scss';

export const PortfolioIcons = ({ title }) => {
  return (
    <div className={styles.bar}>
      <div className={styles.icons}>
        <div className={styles.icon} />
        <div className={styles.icon} />
        <div className={styles.icon} />
      </div>
      <div className={styles.tab}>{title}</div>
    </div>
  );
};

export default PortfolioIcons;

PortfolioIcons.propTypes = {
  title: PropTypes.string.isRequired,
};
