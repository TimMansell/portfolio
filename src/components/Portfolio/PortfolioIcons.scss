$icon-colour-primary: #ff5f58;
$icon-colour-secondary: #ffbd2e;
$icon-colour-tertiary: #28ca42;

.portfolio-icons {
	display: flex;
	padding: 10px;

	&__icon {
		width: 10px;
		height: 10px;
		border-radius: 50%;
		margin-right: 8px;

		&:first-of-type {
			background: $icon-colour-primary;
			box-shadow: inset 0 0 1px darken($icon-colour-primary, 25%);
		}

		&:nth-of-type(2) {
			background: $icon-colour-secondary;
			box-shadow: inset 0 0 1px darken($icon-colour-secondary, 25%);
		}

		&:last-of-type {
			background: $icon-colour-tertiary;
			box-shadow: inset 0 0 1px darken($icon-colour-tertiary, 25%);
		}
	}
}