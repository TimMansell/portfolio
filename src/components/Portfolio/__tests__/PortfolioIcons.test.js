import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom';

import PortfolioIcons from '../PortfolioIcons';

const props = {
  title: 'test title',
};

describe('PortfolioIcons', () => {
  it('should render my component', () => {
    render(<PortfolioIcons {...props} />);
  });

  it('should match snapshot', () => {
    const { asFragment } = render(<PortfolioIcons {...props} />);

    expect(asFragment()).toMatchSnapshot();
  });
});
