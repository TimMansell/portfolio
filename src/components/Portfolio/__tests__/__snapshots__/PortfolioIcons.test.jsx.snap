// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`PortfolioIcons > should match snapshot 1`] = `
<div
  className="_bar_a41ad0"
>
  <div
    className="_icons_a41ad0"
  >
    <div
      className="_icon_a41ad0"
    />
    <div
      className="_icon_a41ad0"
    />
    <div
      className="_icon_a41ad0"
    />
  </div>
  <div
    className="_tab_a41ad0"
  >
    test title
  </div>
</div>
`;
